lib/dev_server.dart

```diff
@@ -84,6 +93,30 @@ class ButterflyDevServer {
     request.response.write(JSON.encode(result));
   }
 
+  Future<Null> _serveDevServerClientFile(HttpRequest request) async {
+    final packagesFile = new File('.packages');
+
+    if (!await packagesFile.exists()) {
+      throw new StateError('.packages file not found. Have you run pub get?');
+    }
+
+    List<String> packageInfoParts = await packagesFile
+        .openRead()
+        .transform(const Utf8Decoder())
+        .transform(const LineSplitter())
+        .map((String line) => line.split(':'))
+        .firstWhere((List<String> parts) => parts[0] == 'butterfly',
+            defaultValue: () => null);
+
+    if (packageInfoParts == null) {
+      throw new StateError('"butterfly" package not found in .packages file. Please check your pubspec.yaml and run pub get again.');
```

Comment:
I'm going for semantics as explained here: https://www.dartlang.org/guides/language/effective-dart/usage#do-throw-objects-that-implement-error-only-for-programmatic-errors

IOW, when this happens there's no attempt to recover, just crash.